code/modules/reagents/Chemistry-Reagents.dm

```diff
@@ -2471,39 +2471,17 @@
 		return 1
 
 	switch(volume)
-		if(1 to 5)
+		if(0.1 to 5)
 			if(ishuman(M))
 				var/mob/living/carbon/human/H = M
 				if(H.species.name != "Diona")
-					var/datum/organ/external/affecting = H.get_organ()
-					for(var/datum/wound/W in affecting.wounds)
-						spawn(1)
-							affecting.wounds -= W
-							H.visible_message("<span class='warning'>[H]'s wounds close up in the blink of an eye!</span>")
-					if(H.getOxyLoss()>0 && prob(90))
-						if(holder.has_reagent("mednanobots"))
-							H.adjustOxyLoss(-4)
-							holder.remove_reagent("mednanobots", 4/40)  //The number/40 means that every time it heals, it uses up number/40ths of a unit, meaning each unit heals 40 damage
-						else
-					if(H.getBruteLoss()>0 && prob(90))
-						if(holder.has_reagent("mednanobots"))
-							H.heal_organ_damage(5, 0)
-							holder.remove_reagent("mednanobots", 5/40)
-						else
-					if(H.getFireLoss()>0 && prob(90))
-						if(holder.has_reagent("mednanobots"))
-							H.heal_organ_damage(0, 5)
-							holder.remove_reagent("mednanobots", 5/40)
-						else
-					if(H.getToxLoss()>0 && prob(50))
-						if(holder.has_reagent("mednanobots"))
-							H.adjustToxLoss(-2)
-							holder.remove_reagent("mednanobots", 2/40)
-						else
-					if(H.getCloneLoss()>0 && prob(60))
+					if(H.getOxyLoss()>0 || H.getBruteLoss()>0 || H.getToxLoss()>0 || H.getFireLoss()>0 || H.getCloneLoss()>0)
 						if(holder.has_reagent("mednanobots"))
-							H.adjustCloneLoss(-2)
-							holder.remove_reagent("mednanobots", 2/40)
+							H.adjustOxyLoss(-5)
+							H.heal_organ_damage(5, 5)
+							H.adjustToxLoss(-5)
+							H.adjustCloneLoss(-5)
+							holder.remove_reagent("mednanobots", 10/40)  //The number/40 means that every time it heals, it uses up number/40ths of a unit, meaning each unit heals 40 damage
```

Comment:
I personally find it strange that medical nanobots wouldn't be able to filter out toxins at a better rate than your standard anti-toxin and heal oxygen damage, also I thought the main point of it was that it would react to damage and lay dormant until you get damaged.
It also performed worse as a physical centric healing chem because of this RNG system since it could at times only heal toxin or oxy for minimal effect while not healing brute or burn.

My biggest gripe was that this hard to get chem was in many cases just as good as regular chems and was only really strong when the chest was damaged, and when the chest was damaged it was borderline overpowered.
In reality nobody makes nanobots mainly because the recipe gives so little and it is a pain to make and the ones who make it dupe it with the Ody and even then its effects are very specialized.

I was thinking of redesigning the recipe, making it slightly easier to make and make it give back more nanobots and to have the Ody drain its charge very rapidly should it try to synthesize medical nanobots.